File: oo-workshop/interface-segregation/src/main/java/com/example/mobprogramming/problem/WalkInClientImpl.java
package com.example.mobprogramming.problem;

public class WalkInClientImpl implements RestaurantInterface{
    public String name;
    public String food;
    public int price;

    public WalkInClientImpl(String name, String food, int price){
        this.name = name;
        this.food = food;
        this.price = price;
    }

    @Override
    public String acceptOnlineOrder() {
        //not applicable for walk-in client
        throw new UnsupportedOperationException();
    }
    @Override
    public String payOnline() {
        //not applicable for walk-in client
        throw new UnsupportedOperationException();
    }
    @Override
    public String walkInCustomerOrder() {
        String message = "Walk-in order from " + this.name + " for " + this.food;
        return message;
    }
    @Override
    public String payInPerson() {
        String message = "In-person payment for $" + this.price + " processed";
        return message;
    }
}

File: oo-workshop/tsexercises/src/interfacesegregation/walkInClientImpl.ts
import {StoreInterface} from './storeInterface'

export class WalkInClientImpl implements StoreInterface{
    //a more complete solution may use the shopping cart from the domain modeling exercise
    public customer: string
    public product: string
    public price: string

    public constructor(name: string, food: string, price: string){
        this.customer = name
        this.product = food
        this.price = price
    }

    public acceptOnlineOrder() : string{
        //not applicable for walk-in client
        throw new Error("not supported")
    }
    
    public payOnline(): string{
        //not applicable for walk-in client
        throw new Error("not supported")
    }
    
    public walkInCustomerOrder(): string{
        const message = "Walk-in order from " + this.customer + " for " + this.product
        return message
    }

    public payInPerson(): string{
        const message = "In-person payment for $" + this.price + " processed"
        return message
    }
}
